cloudflare/resource_cloudflare_zone_settings_override.go

```diff
@@ -689,10 +693,13 @@ func updateSingleZoneSettings(zoneSettings []cloudflare.ZoneSetting, client *clo
 func updateUniversalSSLSetting(zoneSettings []cloudflare.ZoneSetting, client *cloudflare.API, zoneID string) ([]cloudflare.ZoneSetting, error) {
 	indexToCut := -1
 	for i, setting := range zoneSettings {
+		// Skipping USSL Update if value is empty, especially when we are reverting to the initial state and we did not had the information
 		if setting.ID == "universal_ssl" {
-			_, err := client.EditUniversalSSLSetting(zoneID, cloudflare.UniversalSSLSetting{Enabled: boolFromString(setting.Value.(string))})
-			if err != nil {
-				return zoneSettings, err
+			if setting.Value.(string) != "" {
```

Comment:
Any reason why we can't have the `if` above this as `if setting.ID == "universal_ssl && setting.Value.(string) != ""` to avoid the extra nesting?

Reply:
In case universal_ssl is an empty string in initial_settings we would not remove the element from the slice, passing it to `client.UpdateZoneSettings` where it's unknown. That could happen when the resource is removed and we did not have initial_settings for this